packages/x-data-grid/src/constants/cssVariables.ts

```diff
@@ -0,0 +1,61 @@
+const spacingUnit = '--DataGrid-t-spacing-unit';
+
+const breakpoints = {
+  // These come from material-ui, they can't be configured by the theme.
+  // We might want to provide an option to configure them globally.
+  values: {
+    xs: 0, // phone
+    sm: 600, // tablet
+    md: 900, // small laptop
+    lg: 1200, // desktop
+    xl: 1536, // large screen
+  },
+  up: (key: any) => {
+    const values = breakpoints.values as any;
+    const value = typeof values[key] === 'number' ? values[key] : key;
+    return `@media (min-width:${value}px)`;
+  },
+};
+
+export const vars = {
+  spacing,
+  spacingUnit,
+
+  breakpoints,
+
+  palette: {
+    background: {
+      default: '--DataGrid-t-palette-background-default',
+      /** Equivalent to "paper", used for elements floating above other ones. */
+      elevated: '--DataGrid-t-palette-background-elevated',
+    },
+    divider: '--DataGrid-t-palette-divider',
+  },
```

Comment:
Here's what I would suggest for the theme shape, should be enough to cover the vars we need right now, and scalable enough for future iterations.

```suggestion
  colors: {
    foreground: {
      base: '--DataGrid-t-colors-foreground-base',
      muted: '--DataGrid-t-colors-foreground-muted',
      disabled: '--DataGrid-t-colors-foreground-disabled',
    },
    background: {
      base: '--DataGrid-t-colors-background-base',
      overlay: '--DataGrid-t-colors-background-overlay',
    },
    interactive: {
      hover: '--DataGrid-t-colors-interactive-hover',
      hoverOpacity: '--DataGrid-t-colors-interactive-hover-opacity',
      focus: '--DataGrid-t-colors-interactive-focus',
      focusOpacity: '--DataGrid-t-colors-interactive-focus-opacity',
      disabled: '--DataGrid-t-colors-interactive-disabled',
      disabledOpacity: '--DataGrid-t-colors-interactive-disabled-opacity',
      selected: '--DataGrid-t-colors-interactive-selected',
      selectedOpacity: '--DataGrid-t-colors-interactive-selected-opacity',
    },
    border: {
      base: '--DataGrid-t-colors-border-base',
    },
  },
  radius: {
    base: '--DataGrid-t-radius-base',
  },
  fontFamily: {
    base: '--DataGrid-t-font-family-base',
  },
  fontWeight: {
    light: '--DataGrid-t-font-weight-light',
    regular: '--DataGrid-t-font-weight-regular',
    medium: '--DataGrid-t-font-weight-medium',
    bold: '--DataGrid-t-font-weight-bold',
  },
  typography: {
    body: {
      fontFamily: '--DataGrid-t-typography-body-font-family',
      fontSize: '--DataGrid-t-typography-body-font-size',
      fontWeight: '--DataGrid-t-typography-body-font-weight',
      letterSpacing: '--DataGrid-t-typography-body-letter-spacing',
      lineHeight: '--DataGrid-t-typography-body-line-height',
    },
    small: {
      fontFamily: '--DataGrid-t-typography-small-font-family',
      fontSize: '--DataGrid-t-typography-small-font-size',
      fontWeight: '--DataGrid-t-typography-small-font-weight',
      letterSpacing: '--DataGrid-t-typography-small-letter-spacing',
      lineHeight: '--DataGrid-t-typography-small-line-height',
    },
  },
  transitions: {
    easing: {
      easeIn: '--DataGrid-t-transitions-easing-ease-in',
      easeOut: '--DataGrid-t-transitions-easing-ease-out',
      easeInOut: '--DataGrid-t-transitions-easing-ease-in-out',
    },
    duration: {
      short: '--DataGrid-t-transitions-duration-short',
      base: '--DataGrid-t-transitions-duration-base',
      long: '--DataGrid-t-transitions-duration-long',
    },
  },
  zIndex: {
    panel: '--DataGrid-t-z-index-panel',
    menu: '--DataGrid-t-z-index-menu',
  },
```

Reply:
No this is just for us to use, I prefer writing `vars.a.b.c` than `vars['a-b-c']`.

Reply:
Users won't be touching this, except once per design system (once for mantine, once for tailwind, etc). This object should never be mutated.